client/state/checklist/reducer.js

```diff
@@ -1,28 +1,58 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * Internal dependencies
  */
 import { combineReducers, keyedReducer } from 'state/utils';
 import {
 	JETPACK_MODULE_ACTIVATE_SUCCESS,
+	JETPACK_MODULE_DEACTIVATE_SUCCESS,
 	SITE_CHECKLIST_RECEIVE,
 	SITE_CHECKLIST_TASK_UPDATE,
 } from 'state/action-types';
 import { items as itemSchemas } from './schema';
 
-const markChecklistTaskComplete = ( state, taskId ) => ( {
+const toggleChecklistTask = ( state, taskId ) => ( {
```

Comment:
I'd prefer to be explicit with the state we're setting with something like:

```js
// Add a boolean completion argument
const setChecklistTaskCompletion = ( state, taskId, isComplete ) => …
```

Toggling requires a coherent start state and generally seems to be fragile.

---

An additional minor concern is that this overwrites an object with a literal object that only has the `completed` property. Presumably other properties could exist on that object and our intention here is not to wipe them out.

Right now, in practice, I don't think any other properties live there, but it does seem like preparing a future bug for introduction.

Reply:
7a62e9d715